with-editor.el

```diff
@@ -115,11 +115,16 @@ Determining an Emacsclient executable suitable for the
 current Emacs instance failed.  For more information
 please see https://github.com/magit/magit/wiki/Emacsclient."))))
 
+(defun with-editor--emacsclient-name ()
+  (if (s-starts-with-p (downcase invocation-name) "remacs")
```

Comment:
Ah, but those are directories, not executables. The actual executable is at `Emacs.app/some/path/emacs`.